src/index.ts

```diff
@@ -13,36 +13,56 @@ import { IKernel, IKernelSpecs } from '@jupyterlite/kernel';
 
 import { WebWorkerKernel } from './web_worker_kernel';
 
-function getJson(url: string) {
-  const json_url = URLExt.join(PageConfig.getBaseUrl(), url);
-  const xhr = new XMLHttpRequest();
-  xhr.open('GET', json_url, false);
-  xhr.send(null);
-  return JSON.parse(xhr.responseText);
-}
+/**
+ * Fetches JSON data from the specified URL asynchronously.
+ *
+ * This function constructs the full URL using the base URL from the PageConfig and
+ * the provided relative URL. It then performs a GET request using the Fetch API
+ * and returns the parsed JSON data.
+ *
+ * @param {string} url - The relative URL to fetch the JSON data from.
+ * @returns {Promise<any>} - A promise that resolves to the parsed JSON data.
+ * @throws {Error} - Throws an error if the HTTP request fails.
+ *
+ */
+async function getJson(url: string) {
+  const jsonUrl = URLExt.join(PageConfig.getBaseUrl(), url);
+  const response = await fetch(jsonUrl, { method: 'GET' });
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
 
-let kernel_list: string[] = [];
-try {
-  kernel_list = getJson('xeus/kernels.json');
-} catch (err) {
-  console.log(`Could not fetch xeus/kernels.json: ${err}`);
-  throw err;
+  const data = await response.json();
+  return data;
 }
 
-const plugins = kernel_list.map((kernel): JupyterLiteServerPlugin<void> => {
-  return {
-    id: `@jupyterlite/xeus-${kernel}:register`,
-    autoStart: true,
-    requires: [IKernelSpecs],
-    optional: [IServiceWorkerManager, IBroadcastChannelWrapper],
-    activate: (
-      app: JupyterLiteServer,
-      kernelspecs: IKernelSpecs,
-      serviceWorker?: IServiceWorkerManager,
-      broadcastChannel?: IBroadcastChannelWrapper
-    ) => {
+const kernelPlugin: JupyterLiteServerPlugin<void> = {
+  id: '@jupyterlite/xeus-kernel:register',
```

Comment:
Looks like having a single plugin would prevent disabling some kernels via `disabledExtensions`?

Reply:
We can keep the current behavior by checking for the `PageConfig.disabledExtensions` value in the activation function?

Reply:
The reason we need synchronous XHR is to generate JupyterLab plugins dynamically from the fetch data. Which I do not feel is necessary since these plugins do not provide any token. If it's just about to disable some kernels later, we can have other mechanisms

Reply:
Right, it would be great to remove this synchronous XHR.

Reply:
Do we really need the ability to disable kernels?

If users don't want a kernel in the UI, their solution is to not install it in the provided `environment.yml` file.